### DISCUSSION TITLE
Error: EROFS: read-only file system, mkdir '/var/task/tmp'

### DISCUSSION THREAD

**Discussion by Raunak0713** [vercel/community#7317] [Help]
### Summary

Hey Im using a simple react and node project but getting an error in deploying it , vercel shows the error Error: EROFS: read-only file system, mkdir '/var/task/tmp', how to resolve it

### Example

https://github.com/Raunak0713/ReactImageBackEnd/blob/main/index.js

### Steps to Reproduce

![Uploading Screenshot 2024-06-28 at 8.03.44 PM.png…]()

**Comment by versecafe** [comment]
@Raunak0713 Vercel is a serverless runtime built on AWS Lambdas which means you can't write to the local filesystem since it would be thrown away as soon as the function returns, as such they make the fs read only.

```ts
const storage = multer.diskStorage({
    destination: './tmp/',
    filename: (req, file, cb) => {
        cb(null, Date.now() + "-" + file.originalname);
    }
});
```

[a comment by an automated account is not shown]
